src/modules/UI/scenes/CryptoExchange/CryptoExchangeSceneComponent.js

```diff
@@ -85,6 +75,10 @@ type State = {
 }
 
 export class CryptoExchangeSceneComponent extends Component<Props, State> {
+  fromAmountNative: string
+  fromAmountDisplay: string
+  toAmountNative: string
+  toAmountDisplay: string
   constructor (props: Props) {
     super(props)
     const newState: State = {
```

Comment:
they are different in the following functions
```launchFromWalletSelector = () => {
    this.props.openModal(Constants.FROM)
    this.setState({
      whichWallet: Constants.FROM
    })
  }

  launchToWalletSelector = () => {
    this.props.openModal(Constants.TO)
    this.setState({
      whichWallet: Constants.TO
    })
  }
```

The whichWalletFocus property has to do with which one of the items is expanded , however whichWallet is used for deciding whichWallet is directly related to launching the modal to pick the wallet for the exchange. 
It is also crucial that they are different when we get to rendering the modal to exclude the currency codes.

```renderDropUp = () => {
    const { onSelectWallet, fromCurrencyCode, fromWallet, toCurrencyCode, toWallet } = this.props
    const { whichWallet } = this.state
    let excludedCurrencyCode = ''
    // some complex logic because 'toCurrencyCode/fromCurrencyCode'
    // can be denomination (needs to change to actual currencyCode)
    if (whichWallet === Constants.TO) {
      if (fromWallet) {
        if (fromWallet.enabledTokens.length > 1) {
          // could be token
          excludedCurrencyCode = fromCurrencyCode
        } else {
          excludedCurrencyCode = fromWallet.currencyCode
        }
      }
    } else {
      if (toWallet) {
        if (toWallet.enabledTokens.length > 1) {
          // could be token
          excludedCurrencyCode = toCurrencyCode
        } else {
          excludedCurrencyCode = toWallet.currencyCode
        }
      }
    }
    if (this.props.showWalletSelectModal) {
      return (
        <WalletListModal
          onSelectWallet={onSelectWallet}
          topDisplacement={Constants.CRYPTO_EXCHANGE_WALLET_DIALOG_TOP}
          type={Constants.CRYPTO_EXCHANGE}
          whichWallet={whichWallet}
          excludedCurrencyCode={excludedCurrencyCode}
        />
      )
    }
    return null
  }
```

Reply:
What is the scenario where they could be different. In the old design it was possible, in the new design, it doesn't seem possible for them to be different.

Reply:
Steps:
1. no wallet is selected default state is 
```
whichWallet: Constants.FROM,
whichWalletFocus: Constants.FROM,
```


2. I press the button to pick my source wallet. When I do that the state is changed but `whichWallet: Constants.FROM,` is the only change

3. After I have picked my from wallet and the modal closes, since the `whichWalletFocus = Constants.FROM` the top component is expanded with a flip input

4. I click on the destination wallet and the state changes `whichWallet = Constants.TO` however the focus does not change. 

5. When I have picked my destination wallet from the modal and return, `whichWallet = Constants.TO` however `whichWalletFocus= Contants.FROM` so that the top component still has the focus and remains expanded while the bottom component is in a collapsed visual state.